### ISSUE TITLE
Loosing network on OVH VPS

### ISSUE THREAD

**Issue by charlycoste** [elitak/nixos-infect#18]
I tried on a [VPS 2016 SSD 3](https://www.ovh.com/fr/vps/vps-ssd.xml) from OVH, each of these : 
- Debian 8
- Debian 9
- Ubuntu 16.04

And each time I run the script, then installation + reboot are going well. But after that, I can't connect to the VPS anymore. I accessed it by KVM to debug and it seems that the VPS just get disconnected from network.

**Comment by elitak** [comment]
The network detection part of the script is not very robust and assumes digitalocean's idiosyncrasies. I'll probably improve it sometime in the future, but for now, try the following to manually provision your hosts:

(if you have console access, do only step 5 and then `nixos-rebuild switch`)
1. copy over the nixos-infect script
2. edit it
3. comment out the last 4 lines (`makeSwap` to `reboot`)
4. run `source nixos-infect; set +e`. This will generate the config files but not try to install everything yet.
5. edit /etc/nixos/networking.nix, correcting any obvious errors. Use commands `ip addr` `ip route` and `cat /etc/resolv.conf` to obtain any missing info. Probably remove eth1 entirely.
6. edit nixos-infect again, and uncomment the lines you commented
7. `bash -x nixos-infect`

Post me the networking.nix contents, if you still can't get it working.

**Comment by kniteli** [comment]
Yeah, for OVH, don't even both with that the networking just rip it out completely, change this:

`  imports = [
    ./hardware-configuration.nix
    ./networking.nix # generated at runtime by nixos-infect
    $NIXOS_IMPORT
  ];
`
to this:

`
  imports = [
    ./hardware-configuration.nix
    $NIXOS_IMPORT
  ];
`

**Comment by elitak** [comment]
I should probably add a `--no-networking` option that does this, or detect when the original system's network uses dhcp instead of manual config.

**Comment by asymmetric** [comment]
Is this still an issue, since we do [this](https://github.com/elitak/nixos-infect/blob/2d3b1dffa98b9f655a7b9c942ac83dbb22cef514/nixos-infect#L15)?

**Comment by charlycoste** [comment]
I'll try to check it out, then tell you if it's okay now.

**Comment by charlycoste** [comment]
@asymmetric Yes, this is still an issue.
But for information, I tried this tutorial and it worked : https://nixos.wiki/wiki/Install_NixOS_on_Online.Net

**Comment by TheSirC** [comment]
I have big issues to make it run on master, does any of you (@asymmetric, @charlycoste, @kniteli ) would have a functioning version ? I could take care of piggy-backing on it to set up a PR for master to function on OVH.

**Comment by asymmetric** [comment]
Don't use OVH, sorry.

**Comment by elitak** [comment]
@TheSirC if you still need help, attach a .log here with the output after you `set -x` and run the script (comment out the reboot), because I have no detail on what your "big issues" are.

**Comment by TheSirC** [comment]
@elitak Yes, of course. Here is the error-log :
<details>
<summary>Error log</summary>

```
1024+0 records in
1024+0 records out
1073741824 bytes (1.1 GB) copied, 11.4978 s, 93.4 MB/s
swapon: /tmp/nixos-infect.rbzUv.swp: found swap signature: version 1d, page-size 4, same byte order
swapon: /tmp/nixos-infect.rbzUv.swp: pagesize=4096, swapsize=1073741824, devsize=1073741824
```
</details>

The problem lies in the fact that _even with `reboot`_ command on the system does not reboot and the instance is not provided with usual NixOS commands (`nix-env`, `nixos-rebuild`, etc). 
I "bisected" that execution arrive to the `makeConf` function and just "don't execute it" (I can not find any trace of the commands in there leaving any traces on the system). The script exits with error code 1.

**Comment by elitak** [comment]
Run `set -x`, before you run the script, for more detail; that should get you the exact line that fails.

**Comment by TheSirC** [comment]
I actually added it to the script itself without further output.
I added to the interactive session with this output : 
```
root@address:~# ./nixos-infect
+ ./nixos-infect
```
And returning the interactive prompt.

**Comment by TheSirC** [comment]
After further testing (and multiple reinstalls of the VPS to make sure to work on a clean system each time) I found that : 
1. the script is stopping [here, on the grep part](https://github.com/elitak/nixos-infect/blob/6c6e63594dd831a9b0177add82c84a4ebbd6f7af/nixos-infect#L12); running the command myself sends back an empty string, totally normal the file is empty _but_ does exist ! (that is considered as a fail for grep: it sends back error code 1).
2. [Here](https://github.com/elitak/nixos-infect/blob/6c6e63594dd831a9b0177add82c84a4ebbd6f7af/nixos-infect#L194) the script does not include a refresh of the packages list (e.g. `apt-get update`) which can make it fail.

Fun fact : 
The following commands do not produce the same output and I _really_ would like to know why : 
1. `bash -x script` (<-- I ran this one to have output on the script)
2. adding `set -x` to the script after the shebang
3. doing `set -x` in your interactive prompt and then running `./script`

**Comment by TheSirC** [comment]
So after applying patches for the above-cited issues I opened [a pull-request](../pull/38) that worked for me on OVH.

**Comment by raspher** [comment]
Resolved since 3 years....

Anyway i've similar problem on different budget provider. It's kinda weird -> vps does respond to ping but seems that  it have all ports closed. Any ideas what's the problem? 

Anyway if i've enought time, i'll try do step by step what this script does and play a little with config. Maybe #61 is the solution?
